### DISCUSSION TITLE
Documentation feedback wanted

### DISCUSSION THREAD

**Discussion by cortinico** [reactwg/react-native-new-architecture#7] [Documentation]
Hi all,
We've created this section to collect feedback on the documentation related to the New Architecture for React Native.

Specifically:
* All the feedback related to the https://reactnative.dev documentation **should be posted on the specific threads**.
  * https://github.com/reactwg/react-native-new-architecture/discussions/8
  * https://github.com/reactwg/react-native-new-architecture/discussions/9
* If you created a blogpost, video, podcast or any other material that is worth sharing related to the New Architecture, **please post it here below**.

**Comment by Productivix** [comment]
thank you Nicolas for accepting me in your group.

I was testing (following the advises on) the page https://reactnative.dev/docs/the-new-architecture/pillars-turbomodules?turbomodule-specs=flow 
and you can see that : 
on https://reactnative.dev/docs/the-new-architecture/pillars-turbomodules?turbomodule-specs=flow#2-javascript-specification
the file NativeCalculator.js is NOT a js file but almost a ts file

**Reply by satya164** [reply]
> the file NativeCalculator.js is NOT a js file but almost a ts file

That's not ts or js, it's flow. flow files have `.js` extension.

**Reply by cortinico** [reply]
As @satya164 answered, those are Flow files and are using the `.js` extension

**Comment by Productivix** [comment]
ok may be , sorry, I do not know, (I just saw .js ...) - thanks 
 :  that appears as error in Vscode.., that is it mainly

**Reply by satya164** [reply]
If you want to write it in flow you'd need to setup flow extension in vscode. though I'd suggest using typescript instead

**Reply by Productivix** [reply]
ok thanks - but sorry never seen a program where Flow is requiered ...
I have TS to work still ..it is enough

**Comment by Productivix** [comment]
I have another thing : 
a correction in https://reactnative.dev/docs/the-new-architecture/pillars-turbomodules?turbomodule-specs=flow&final-app=typescript#javascript 
(the app.tsx)

- :   <Text style={{marginLeft: 20, marginTop: 20}}>
+ : <Text style={{marginLeft: 20, marginTop: 20,color:'white'}}>
because we cannot see on dark the text

**Reply by Productivix** [reply]
or 'grey' (because I activated dark theme on my terminal)

**Comment by Productivix** [comment]
very good it works ! Thanks ! this method is simpler than  the one on https://reactnative.dev/docs/native-modules-setup that bugs on Linux !

**Comment by noway** [comment]
Hi @cortinico had trouble finding documentation on 2 questions:

- Has React Native team made a commitment to support Old Architecture indefinitely? 
- If not, when does React Native team expect Old Architecture support to stop? Approximate RN version or approximate calendar year.

I understand it may be too early to tell, but we're currently figuring out when to schedule work to adopt New Architecture internally. Having an approximate timeline would help us prioritize. Thanks!

**Reply by cortinico** [reply]
> Has React Native team made a commitment to support Old Architecture indefinitely?

We'll be supporting the Old Architecture in the immediate future, as long as the two Architecture can be used side by side.

As you can image, we're looking into sunsetting the Old Architecture, so you can expect the support on that end to slowly fade away. For example, we're prioritising PRs and Issues related to the New Architecture at the moment, so a bug on the old renderer will generally take longer to be fixed.

> If not, when does React Native team expect Old Architecture support to stop? Approximate RN version or approximate calendar year.

We don't have an exact that about the end of support for Old Architecture. But I can mention that if you consider migrating to the New Architecture, and open a new issue on it, that will get a higher priority on our backlog than anything old arch.
This is specifically true now as we're looking closely into New Architecture migration. It won't be the same in the future.

**Reply by ptoro** [reply]
Hi @cortinico. I'd love to hear if there has been any additional clarity gained on noway's second bullet in the last year. An approximate RN version and/or calendar year would help a lot as we make update decisions.

Also, about how much notice should we expect once an exact support timeline is announced before support for the old architecture is dropped (e.g. 3 months, 6 months)?
